build/scripts/install_externals.py

```diff
@@ -0,0 +1,207 @@
+import os
+import json
+import requests
+import zipfile
+import shutil
+from enum import Enum
+
+class Host(Enum):
+    UNKNOWN = 0
+    GITLAB = 1
+    GITHUB = 2
+
+def __get_script_path():
+    return os.path.dirname(os.path.realpath(__file__))
+def __get_root_path():
+    root = __get_script_path()
+    root = os.path.join(root, os.path.join("..", ".."))
+
+    return root
+
+def __get_host(path):
+    if "gitlab" in path:
+        return Host.GITLAB
+    elif "github" in path:
+        return Host.GITHUB
+    
+    print("Unknown host!")
+    return Host.UNKNOWN
+
+def __build_gitlab_path(baseUrl, name, tag):
+    url = os.path.join(baseUrl, "-")
+    url = os.path.join(url, "archive")
+    url = os.path.join(url, tag)
+    url = os.path.join(url, name+"-"+tag+".zip")
+    url = url.replace("\\", "/")
+    return url
+def __build_github_path(baseUrl, tag):
+    url = os.path.join(baseUrl, "archive")
+    url = os.path.join(url, "refs")
+    url = os.path.join(url, "tags")
+    url = os.path.join(url, tag+".zip")
+    url = url.replace("\\", "/")
+    return url
+def __build_host_path(baseUrl, name, tag):
+    host = __get_host(baseUrl)
+    match host:
```

Comment:
match is only supported in python 3.10, a lot of people won't be able to run this